kerastuner/tuners/bayesian.py

```diff
@@ -0,0 +1,285 @@
+import numpy as np
+import random
+import json
+from scipy import optimize as scipy_optimize
+from sklearn import gaussian_process
+
+from ..abstractions.tensorflow import TENSORFLOW_UTILS as tf_utils
+from ..engine import tuner as tuner_module
+from ..engine import oracle as oracle_module
+from ..engine import hyperparameters as hp_module
+
+
+class BayesianOptimizationOracle(oracle_module.Oracle):
+    """Bayesian optimization oracle.
+
+    It uses Bayesian optimization with a underlying Gaussian process model.
+    The acquisition function used is upper confidence bound (UCB).
+
+    Attributes:
+        init_samples: Int. The number of randomly generated samples as initial
+            training data for Bayesian optimization.
+        alpha: Float. Value added to the diagonal of the kernel matrix
```

Comment:
The docstring needs information about what the user is supposed to do with the argument (and if the user isn't supposed to touch it, it shouldn't be an argument). For instance, below we have "The larger it is, the more explorative it is.", which is useful. Docstrings should be actionable.